### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by romain-grecourt**
Finalize the windows support with a fully working pipeline passing all the unit tests, as well as a native-image windows build.

### CODE REVIEW

**Review by batsatt — CHANGES_REQUESTED** [review]

File: helidon-linker/src/test/java/io/helidon/linker/ClassDataSharingTest.java [hunk 1]
```diff
@@ -49,7 +60,7 @@ void testQuickstartMp() throws Exception {
                                                .jri(JAVA_HOME)
                                                .applicationJar(mainJar)
                                                .createArchive(false)
-                                               .logOutput(false)
+                                               .logOutput(true)
```
[the remark this answers is not in the document]
Comment (romain-grecourt) [reply helidon-linker/src/test/java/io/helidon/linker/ClassDataSharingTest.java:63]: I've had a hard time with these and this was initially intended for debugging but I think we should leave it on. Would that be ok ?

File: utils/src/main/java/io/helidon/build/util/SourcePath.java [hunk 2]
```diff
@@ -241,7 +252,7 @@ public static boolean matches(String[] path, String[] pattern) {
      * Matches the given value with a pattern that may contain wildcard(s)
      * character that can filter any sub-sequence in the value.
      *
-     * @param val the string to filter
+     * @param val     the string to filter
```
[the remark this answers is not in the document]
Comment (romain-grecourt) [reply utils/src/main/java/io/helidon/build/util/SourcePath.java:255]: This is the javadoc formatting done by intellij, it aligns the columns for the param descriptions

[the remark this answers is not in the document]
Comment (romain-grecourt) [reply utils/src/main/java/io/helidon/build/util/SourcePath.java:255]: Nevertheless, I'm removing that change to remove the noise in the diffs

File: helidon-linker/src/test/java/io/helidon/linker/ClassDataSharingTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (batsatt) [reply helidon-linker/src/test/java/io/helidon/linker/ClassDataSharingTest.java:63]: Sure, just wanted to make sure it was intentional.

**Review by batsatt — APPROVED** [review]
LGTM

**Pull request merged**